metadata/src/main/java/org/apache/kafka/controller/BrokersToIsrs.java

```diff
@@ -139,10 +140,16 @@ public TopicIdPartition next() {
      * Partitions with no isr members appear in this map under id NO_LEADER.
      */
     private final TimelineHashMap<Integer, TimelineHashMap<Uuid, int[]>> isrMembers;
+    
+    private final Map<Uuid, Integer> offlinePartitionCounts;
```

Comment:
We cannot use regular maps here because they will not roll back to the desired state during a snapshot restore.

In any case, I don't see why we need this map.  It's enough to know how many offline partitions there are, which we already have a count of below.

Reply:
This was so that when a topic is removed, any offline partitions for that topic are decremented from the counter.

Reply:
All the information that is needed is already here.